### ISSUE TITLE
Rethink mergedicts logic

### ISSUE THREAD

**Issue by cliffckerr** [sciris/sciris#320]
This can lead to subtle bugs, e.g. if the dict-like object requires an input argument:
```py
    # Try to get the output type from the first argument, but revert to a standard dict if that fails
    try:
        assert isinstance(args[0], dict)
        outputdict = args[0].__class__() # This creates a new instance of the class
    except:
        outputdict = {}
```

**Comment by cliffckerr** [comment]
I think the right solution is to separate the assert and the `class` statements so the former can fail and the latter can't, then add a kwarg to not force type consistency

**Comment by cliffckerr** [comment]
Closed by #345
